ally/share.php

```diff
@@ -12,6 +12,33 @@
 		<link rel="stylesheet" type="text/css" href="css/styles.css"/>
 	</head>
 	<body>
+
+<!--script for populating the destination form via url data -->
+		<script type="text/javascript">
+			function getParameterByName(name, url) {
+    			if (!url) {
+    			  url = window.location.href;
+    			}
+    			name = name.replace(/[\[\]]/g, "\\$&");
+    			var regex = new RegExp("[?&]" + name + "(=([^&#]*)|&|#|$)"),
+    			    results = regex.exec(url);
+    			if (!results) return null;
+    			if (!results[2]) return '';
+    			return decodeURIComponent(results[2].replace(/\+/g, " "));
+			}
+
+        function addDestination() {
```

Comment:
I moved this function to the end so that it is run after all the other code and after the page renders